### PR TITLE
{system}[system/system] OpenSSL v3.0

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)

Depends on
- [x] https://github.com/easybuilders/easybuild-easyblocks/pull/2746
- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/18896

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2746
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
broadwell-alma9-eb - Linux AlmaLinux 9.0 (Emerald Puma), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 510.73.08, Python 3.9.10
See https://gist.github.com/6d6ff413baf2377faa2fa9f2f1f4f7d8 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2746
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
eb-centosstream9.novalocal - Linux CentOS Stream 9, x86_64, Intel Xeon Processor (Skylake, IBRS) (skylake_avx512), Python 3.9.10
See https://gist.github.com/1df097501babf31de687bafde5791d57 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2746
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
node080.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/aba24e87c922dcb6bcf26d1baa99a9a1 for a full test report.

**Comment by jfgrimm** [comment]
@SebastianAchilles I'm seeing: `Can't locate IPC/Cmd.pm in @INC`

**Comment by SebastianAchilles** [comment]
> @SebastianAchilles I'm seeing: `Can't locate IPC/Cmd.pm in @INC`

Yes, in the container I have installed ` perl-IPC-Cmd` and `perl-Pod-Html` to build OpenSSL v3.0:

https://github.com/easybuilders/easybuild-containers/blob/main/almalinux-9.0/Dockerfile#L22-L23

Either we could add this as `osdependencies` or we could build a `Perl-minimal` in the `system` toolchain. Which solution do you think would be better?

**Comment by jfgrimm** [comment]
> > @SebastianAchilles I'm seeing: `Can't locate IPC/Cmd.pm in @INC`
> 
> Yes, in the container I have installed ` perl-IPC-Cmd` and `perl-Pod-Html` to build OpenSSL v3.0:
> 
> https://github.com/easybuilders/easybuild-containers/blob/main/almalinux-9.0/Dockerfile#L22-L23
> 
> Either we could add this as `osdependencies` or we could build a `Perl-minimal` in the `system` toolchain. Which solution do you think would be better?

I'd probably go for the latter?

**Comment by boegel** [comment]
Yeah, Perl should be fairly robust to build in multiple OS configurations, I would expect.

But we should test it extensively first, so please open a separate PR for a minimal `Perl` with `system` (and mention in what context we plan to use it).

**Comment by boegel** [comment]
Worth taking into account in this context: https://github.com/easybuilders/easybuild-easyconfigs/issues/8973

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bwd-rockylinux-92 - Linux Rocky Linux 9.2 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), Python 3.9.16
See https://gist.github.com/SebastianAchilles/6063954e39c1ffe631d8c9ff146518c1 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
jsczen3g1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.3, x86_64, AMD EPYC-Milan Processor (zen3), 1 x NVIDIA NVIDIA A100 80GB PCIe, 545.23.08, Python 3.9.18
See https://gist.github.com/SebastianAchilles/703342ec323df0806e88660145514363 for a full test report.

File: easybuild/easyconfigs/o/OpenSSL/OpenSSL-3.0.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+easyblock = 'EB_OpenSSL_wrapper'
+
+name = 'OpenSSL'
+version = '3.0'
```
Comment (Micket) [inline easybuild/easyconfigs/o/OpenSSL/OpenSSL-3.0.eb:4]: I think we should just have `3` here as they will start following semver going foward.

File: easybuild/easyconfigs/o/OpenSSL/OpenSSL-3.0.eb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# this version of OpenSSL, the installation directory of this module will be
+# populated with symlinks to the system files. The minimum required version of
+# OpenSSL can be finely controled with 'minimum_openssl_version' (defaults to
+# easyconfig version).
+# If the host system does not have this version of OpenSSL (or with the option
+# wrap_system_openssl = False), EasyBuild will fall back to the following
+# component list, which will be build and installed as usual.
+
+components = [
+    (name, '3.0.11', {
```
Comment (Micket) [inline easybuild/easyconfigs/o/OpenSSL/OpenSSL-3.0.eb:31]: and maybe we want to use 3.2.x here?

**Review by Micket — CHANGES_REQUESTED** [review]
I saw there were forward looking packages in RHEL8
openssl3.x86_64
openssl3-libs.x86_64
openssl3-devel.x86_64


I'll try it out to see that it actually finds and wraps those

**Comment by boegel** [comment]
Closing this, since we're favoring `OpenSSL/3` (cfr. #19623), since OpenSSL 3.x promises a stable ABI (couldn't find a good reference for that though)

**Pull request closed without merging**